### ISSUE TITLE
Clarification regarding "uint32" in mersenne-twister python code snippet

### ISSUE THREAD

**Issue by Zero-1729** [crypto101/book#363]
Greetings :wave:,

On page 180, where the mersenne-twister initialization function is discussed, there seems to be a `uint32` function call in the python code snippet:-

```py
def initialize_state(seed):
    state = [seed]
    
        for i in xrange(1, 624):
            prev = state[-1]
            elem = 0x6c078965 * (prev ^ (prev » 30)) + i
            state.append(uint32(elem))

    return state
```

In previous code snippets all non built-in functions seem to be explicitly defined. 

It appears as though there may be need for some clarification on the source of this function. At first glance it looks like it could be the `c_uint32` function from the `ctypes` module, I.e.

```py
from ctypes import c_uint32 as uint32
```

However, that changes the element's type to `c_uint`. Which results in the following error:-

```py
TypeError: unsupported operand type(s) for >>: 'c_uint' and 'int'
```

that is, unless the code snippet is modified:-

```py
def initialize_state(seed):
    # ...
    
        for i in xrange(1, 624):
            # ...
            state.append(uint32(elem).value)

    return state
```

Another way could be that it was a user defined function, that is:-

```py
def uint32(n):
    return 0xFFFFFFFF & n
```

At this point this is all speculation on my side. It would be better to hear the Author's perspective on this.

**Comment by lvh** [comment]
Huh, I could've sworm that function was defined somewhere, but yeah that final definition looks right to me.

**Comment by lvh** [comment]
(I'd be happy to take a patch if you're offering :))

**Comment by Zero-1729** [comment]
Sure, I just opened up a PR.

Seeing that the `uint32` call is intentional, I  just defined the `uint32` function above the initial call (in 
 the `initialize_state` function) of the code snippet.

**Comment by lvh** [comment]
Closed via #364.
